R/create_template.R

```diff
@@ -667,24 +667,94 @@ create_template <- function(
+                 paste0(resdir, "/", model_results)
+                 ), "') \n",
           "# Call reference points and quantities below \n",
-          "# start_year <- min(output$year) \n",
-          "# end_year <- '", year, "' \n",
-          "# Fend <- output$estimate[output$label == 'fishing_mortality' & output$year == year] \n",
-          "# Ftarg ", "\n",
-          "# F_Ftarg ", "\n",
-          "# Bend ", "\n",
-          "# Btarg ", "\n",
-          "# tot_catch", "\n",
-          "# sbtarg", "\n",
-          "# M ", "\n",
-          "# Bmsy ", "\n",
-          "# SBmsy ", "\n",
-          "# steep ", "\n",
-          "# R0", "\n",
-          "# fSB ", "\n"
+          "output <- output |> \n",
+          "  ", "dplyr::mutate(estimate = as.numeric(estimate), \n",
+          "  ", "  ", "uncertainty = as.numeric(uncertainty)) \n",
+
+          "start_year <- as.numeric(min(output$year, na.rm = TRUE)) \n",
+          # change end year in the fxn to ifelse where is.null(year)
+          "end_year <- (output |> \n",
+          "  ", "dplyr::filter(!(year %in% c('Virg', 'Init', 'S/Rcurve', 'INIT')), \n",
+          "  ", "  ", "!is.na(year)) |> \n",
+          "  ", "dplyr::mutate(year = as.numeric(year)) |> \n",
+          "  ", "dplyr::summarize(max_val = max(year)) |> \n",
+          "  ", "dplyr::pull(max_val))-10", "\n",
+          # for quantities - don't want any values that are split by factor
+          "output2 <- output |> \n",
+          "  ", "dplyr::filter(is.na(season), \n",
+          "  ", "  ", "is.na(fleet), \n",
+          "  ", "  ", "is.na(sex), \n",
+          "  ", "  ", "is.na(area), \n",
+          "  ", "  ", "is.na(growth_pattern), \n",
+          "  ", "  ", "is.na(subseason), \n",
+          "  ", "  ", "is.na(age))",  "\n",
+
+          "Fend <- output2 |> ", "\n",
+          "  ", "dplyr::filter(c(label == 'fishing_mortality' & year == end_year) | c(label == 'terminal_fishing_mortality' & is.na(year))) |>", "\n",
+          "  ", "dplyr::pull(estimate)", "\n",
+
+          "Ftarg <- output2 |>", "\n",
+          "  ", "dplyr::filter(grepl('f_target', label) | grepl('f_msy', label) | c(grepl('fishing_mortality_msy', label) & is.na(year))) |>", "\n",
+          "  ", "dplyr::pull(estimate)", "\n",
+
+          "F_Ftarg <- Fend / Ftarg", "\n",
+
+          "Bend <- output2 |>", "\n",
+          "  ", "dplyr::filter(grepl('mature_biomass', label) | grepl('^biomass$', label),", "\n",
+          "  ", "  ", "year == end_year) |>", "\n",
+          "  ", "dplyr::pull(estimate)", "\n",
+
+          "Btarg <- output2 |>", "\n",
+          "  ", "dplyr::filter(c(grepl('biomass', label) & grepl('target', label) & estimate >1) | label == 'biomass_msy') |>", "\n",
+          "  ", "dplyr::pull(estimate)", "\n",
+
+          "total_catch <- output |>", "\n",
+          "  ", "dplyr::filter(grepl('^catch$', label), \n,",
```

Comment:
There's an extra comma here

Reply:
Just pushed the change earlier... must have forgot to last night